### ISSUE TITLE
gateway-proxy container throwing libprotobuf errors

### ISSUE THREAD

**Issue by totallyGreg** [solo-io/gloo#7903]
### Gloo Edge Version

1.13.x (latest stable)

### Kubernetes Version

1.20.x

### Describe the bug

Gloo Edge version 1.13.7 

gateway-proxy container in gateway pod logs below error thousands of times a day in our Dev environment: 
```
[libprotobuf ERROR external/com_google_protobuf/src/google/protobuf/wire_format_lite.cc:581] String field 'google.protobuf.Value.string_value' contains invalid UTF-8 data when serializing a protocol buffer. Use the 'bytes' type if you intend to send raw bytes.
```

### Steps to reproduce the bug

Settings 1.13.7 CRD

```
spec:
  consoleOptions:
    apiExplorerEnabled: true
    readOnly: false
  discovery:
    fdsMode: WHITELIST
    fdsOptions:
      graphqlEnabled: true
    udsOptions:
      enabled: false
  discoveryNamespace: gloo-system
  extauth:
    extauthzServerRef:
      name: extauth
      namespace: gloo-system
    requestTimeout: 25s
    transportApiVersion: V3
    userIdHeader: x-user-id
  extensions:
    configs:
      extAuth:
        signingKey:
          key: xxxxxxxxxx
  gateway:
    enableGatewayController: true
    isolateVirtualHostsBySslConfig: false
    readGatewaysFromAllNamespaces: false
    validation:
      allowWarnings: true
      alwaysAccept: false
      disableTransformationValidation: false
      proxyValidationServerAddr: gloo:9988
      serverEnabled: true
      validationServerGrpcMaxSizeBytes: 4000000
      warnRouteShortCircuiting: false
  gloo:
    awsOptions:
      fallbackToFirstFunction: true
    disableKubernetesDestinations: true
    disableProxyGarbageCollection: false
    enableRestEds: false
    invalidConfigPolicy:
      invalidRouteResponseBody: Gloo Gateway has invalid configuration. Administrators
        should run `glooctl check` to find and fix config errors.
      invalidRouteResponseCode: 404
      replaceInvalidRoutes: true
    proxyDebugBindAddr: 0.0.0.0:9966
    regexMaxProgramSize: 200
    restXdsBindAddr: 127.0.0.1:9998
    xdsBindAddr: 127.0.0.1:9999
  graphqlOptions:
    schemaChangeValidationOptions:
      processingRules: []
      rejectBreakingChanges: false
  kubernetesArtifactSource: {}
  kubernetesConfigSource: {}
  kubernetesSecretSource: {}
  observabilityOptions:
    grafanaIntegration:
      defaultDashboardFolderId: 66
  ratelimitServer:
    rateLimitBeforeAuth: false
    ratelimitServerRef:
      name: rate-limit
      namespace: gloo-system
  refreshRate: 60s
  watchNamespaces:
```

### Expected Behavior

Not seeing thousands of errors/day in the logs. 

### Additional Context

I will link ticket that contains customers config-dump

```
glooctl proxy served-config
{"level":"error","ts":"2023-02-22T12:08:45.755-0500","caller":"xdsinspection/get_last_config.go:88","msg":"connecting to gloo failed with err endpoints err: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp [::1]:9999: connect: connection refused\"","stacktrace":"github.com/solo-io/gloo/projects/gloo/cli/pkg/xdsinspection.GetGlooXdsDump\n\t/workspace/gloo/projects/gloo/cli/pkg/xdsinspection/get_last_config.go:88\ngithub.com/solo-io/gloo/projects/gloo/cli/pkg/cmd/gateway.printGlooXdsDump\n\t/workspace/gloo/projects/gloo/cli/pkg/cmd/gateway/served-config.go:31\ngithub.com/solo-io/gloo/projects/gloo/cli/pkg/cmd/gateway.servedConfigCmd.func1\n\t/workspace/gloo/projects/gloo/cli/pkg/cmd/gateway/served-config.go:18\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902\nmain.main\n\t/workspace/gloo/projects/gloo/cli/cmd/main.go:11\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
unable to parse yaml%
```

**Comment by totallyGreg** [comment]
New info in ticket suggests it's related to https://github.com/solo-io/solo-projects/issues/4537

**Comment by jbohanon** [comment]
I am so far unable to reproduce this issue. Additional details such as whether the upstream lambda is ever executed and the full envoy logs near a failing request would be helpful. I am trying to determine if this is due to an interaction between the Lambda filter and the ModSecurity filter, or if the Lambda portion is a red herring.

**Comment by totallyGreg** [comment]
customer updated ticket with debug, lambda and gateway log.

**Comment by jbohanon** [comment]
This issue was  resolved with updates to the ModSecurity filter released with Gloo Edge Enterprise v1.13.10
